### PR TITLE
Added Sub Sequence Kernel cookbook page.

### PR DESCRIPTION
**Pull request by arianepaola**
Added Sub Sequence Kernel cookbook page.

### CODE REVIEW

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+=========================
+String Subsequence Kernel
+=========================
+
+The String Subsequence Kernel (SSK) is discussed by :cite:`lodhi2002ssk`.
+
+A subsequence is any ordered sequence of :math:`n` characters occurring in
+the text, though not necessarily contiguous.
+
+More formally, string :math:`u` is a subsequence of string :math:`s`, :math:`iff` there
```
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: Too much detail. The cookbook examples should only cover basics (in particular for kernels) i.e. only one equation or so. Preferably only use math symbols for things that are later used in the code.

"..... This kernel is an inner product in the feature space generated by all sub-sequences of length n. The sub-sequences are weighted by the exponentially decaying factor λ. More weight is given to those occurrences that are nearly contiguous. The implementation uses an efficient dynamic programming strategy."

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+since the dimension of the feature space grows exponentially with :math:`n`.
+
+The paper describes an efficient computation approach using a dynamic
+programming technique.
+ 
+-------
+Example
+-------
+
+We start by creating
```
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: ehm, this is empty. Did you notice?

File: examples/meta/src/kernel/sub_sequence_kernel.sg [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+CSVFile f_labels_train("../../data/label_train_dna.dat")
+
+#![create_features]
+StringCharFeatures features_train(f_feats_train)
+StringCharFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+#![create_features]
+
+#![create_appropriate_kernel]
+real maxlen = 1
```
Comment (karlnapf) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: if this is called maxlen, this kind of is contrasting the math that says subsequences of exactly n

File: examples/meta/src/kernel/sub_sequence_kernel.sg [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+CSVFile f_feats_test("../../data/fm_test_dna.dat")
+CSVFile f_labels_train("../../data/label_train_dna.dat")
+
+#![create_features]
+StringCharFeatures features_train(f_feats_train)
+StringCharFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+#![create_features]
+
+#![create_appropriate_kernel]
```
Comment (karlnapf) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: minor: just name this create_kernel, why have "appropriate" in the tag

File: examples/meta/src/kernel/sub_sequence_kernel.sg [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#![create_instance]
+
+#![train_and_apply]
+svm.train()
+BinaryLabels labels_predict = svm.apply_binary(features_test)
+#![train_and_apply]
+
+#![evaluate_accuracy]
+AccuracyMeasure eval()
+real accuracy = eval.evaluate(labels_predict, labels_test)
```
Comment (karlnapf) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: labels_test doesnt seem to be defined anywhere

**Comment by karlnapf** [comment]
This is quite incomplete as of now, wont be able to run on travis

**Comment by karlnapf** [comment]
Any updates here?

**Comment by arianepaola** [comment]
@vigsterkr @karlnapf @lgoetz I have mentioned on IRC that I saw the comments and will push an update shortly. Thanks!

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst (hunk 1 above)
Comment (arianepaola) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: @karlnapf I will update this part, but the original text and explanation comes from the Shogun implementation header file.

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst (hunk 2 above)
Comment (arianepaola) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: thanks :+1:

File: examples/meta/src/kernel/sub_sequence_kernel.sg (hunk 3 above)
Comment (arianepaola) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: @karlnapf This was translated from the Python code.

File: examples/meta/src/kernel/sub_sequence_kernel.sg (hunk 4 above)
Comment (arianepaola) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: @karlnapf I will rename it. It was in the example cookbooks.

File: examples/meta/src/kernel/sub_sequence_kernel.sg (hunk 5 above)
Comment (arianepaola) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: Thanks :+1:

**Comment by arianepaola** [comment]
@karlnapf How to add the latter part for ground truth labels?

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst (hunk 1 above)
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: I know, it is a good point, but for now we want to keep the cookbooks a bit more minimal

File: doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst (hunk 2 above)
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/sub_sequence_kernel.rst]: ping me once you filled it

File: examples/meta/src/kernel/sub_sequence_kernel.sg (hunk 3 above)
Comment (karlnapf) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: Part of this task is to clean up mistakes that others did in the past ;)

File: examples/meta/src/kernel/sub_sequence_kernel.sg (hunk 5 above)
Comment (karlnapf) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]: always make sure your meta examples are executable locally before sending a PR with them.
If it bothers you to build the modular ones, at least the cpp should compile and run

**Comment by karlnapf** [comment]
ground truth:
open the train data and labels file and see whether the pattern is easy. If so, just create a test labels file and send a PR to shogun data
Let me know if you need help on this

File: examples/meta/src/kernel/sub_sequence_kernel.sg [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+#![create_features]
+StringCharFeatures features_train(f_feats_train)
+StringCharFeatures features_test(f_feats_test)
+BinaryLabels labels_train(f_labels_train)
+BinaryLabels labels_test(f_labels_test)
+#![create_features]
+
+#![create_kernel]
+real len = 1
+real decay = 1
```
Comment (sorig) [inline examples/meta/src/kernel/sub_sequence_kernel.sg]:
```
real len = 1.0
real decay = 1.0
```
See #3410

**Pull request closed without merging**